### DISCUSSION TITLE
Which virtualization library is recommended?

### DISCUSSION THREAD

**Discussion by hyperknot** [clauderic/dnd-kit#1372] [Q&A]
I see the examples are using https://github.com/clauderic/react-tiny-virtual-list, which is also by @clauderic, but hasn't been updated for 5 years.

I looked for alternatives and found 2:
https://github.com/bvaughn/react-window by @bvaughn
and
https://github.com/inokawa/virtua by @inokawa

What I like about react-window is that I've been using react-sortable-hoc + react-virtualized in production for 5 years, which were the previous libraries by @bvaughn and @clauderic. They are also extremely popular and well tested.

Having said that, virtua also looks nice based on their comparison section.

Does anyone have any experience with virtualization libraries and dnd-kit?

**Comment by inokawa** [comment]
Hello, I'm author of virtua.
dnd-kit + virtua is a recommended set from me. I actually use it in my day time job!

https://inokawa.github.io/virtua/?path=/story/advanced-with-dnd-kit--default
https://github.com/inokawa/virtua/blob/main/stories/react/advanced/With%20dnd-kit.stories.tsx

**Reply by hyperknot** [reply]
Thanks a lot! Indeed, based on your official example the integration is indeed very straighforward!

**Reply by remusris** [reply]
Hi I found a bug with that implementation.  If you have a nested component like a <textarea/> but with auto resizing functionality the dnd-begins to break.  I'm only able to get dnd working when removing the virtua virtualizer.  I did try other DnD libraries and other virtualizers and seemed to run  into the same issue.  

Put in this autoresizing textarea component inside the list entry for the sortable list and you should be able to experience this issue

https://github.com/Andarist/react-textarea-autosize

Appreciate the work you've done with virtua and I'd be even more grateful if you could figure this out.  Basically if you try invoking the dnd-kit autoscroller you should very quickly see the issue that I'm referring to.  

https://github.com/user-attachments/assets/42d591a3-5f71-45cd-b822-37d2ead8140c

**Reply by inokawa** [reply]
Thanks. It might be a similar situation to https://github.com/inokawa/virtua/discussions/588 . Further investigation is needed.

**Comment by corocoto** [comment]
Does anyone have experience with @dnd-kit and react-window?

**Reply by corocoto** [reply]
If anyone want to find example of using - I found it inside the @dnd-kit examples: https://github.com/clauderic/dnd-kit/blob/experimental/apps/stories/stories/react/Sortable/Virtualized/ReactWindowExample.tsx
